plugins/out_s3/s3.c

```diff
@@ -960,9 +984,23 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
         c = mock_s3_call("TEST_PUT_OBJECT_ERROR", "PutObject");
     }
     else {
-        c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
-                                              uri, body, body_size,
-                                              NULL, 0);
+        if (ctx->canned_acl == NULL) {
+            c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
+                                                  uri, body, body_size,
+                                                  NULL, 0);
+        }
+        else {
+            canned_acl_header = create_canned_acl_header(ctx->canned_acl);
+            if (canned_acl_header == NULL) {
+                flb_sds_destroy(uri);
+                flb_plg_error(ctx->ins, "Failed to get the canned ACL policy for objects");
```

Comment:
Let's do "Failed to create canned ACL header"

"Failed to get" sounds a bit off, we got the ACL from the user